core/src/transaction_status_service.rs

```diff
@@ -99,8 +99,9 @@ impl TransactionStatusService {
                     })
                     .expect("FeeCalculator must exist");
                 let fee = fee_calculator.calculate_fee(transaction.message());
-                let (writable_keys, readonly_keys) =
-                    transaction.message.get_account_keys_by_lock_type();
+                let (writable_keys, readonly_keys) = transaction
+                    .message
+                    .get_account_keys_by_lock_type(bank.demote_sysvar_write_locks());
```

Comment:
@CriesofCarrots @mvines I think this will affect how it shows in the explorer. Should we show sysvar as demoted to read locked even if it is write in the transaction? or as the original raw write lock?

Reply:
My preference would be to show sysvar accounts as they were processed, so demoted to read-locked.